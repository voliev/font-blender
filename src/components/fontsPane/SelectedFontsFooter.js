import React from "react";
import PropTypes from "prop-types";
import "./styles/SelectedFontsFooter.css";

const SelectedFontsFooter = ({ clearSelectedFontsButtonClickHandler }) => (
  <div className="SelectedFontsFooter">
    <button
      className="btn btn-no-background"
      onClick={clearSelectedFontsButtonClickHandler}
      title="Clear selected fonts"
      value="clear"
    >
      Clear
    </button>
  </div>
);

SelectedFontsFooter.propTypes = {
  clearSelectedFontsButtonClickHandler: PropTypes.func.isRequired
};

export default SelectedFontsFooter;
