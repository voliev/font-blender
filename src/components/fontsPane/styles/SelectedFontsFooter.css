.SelectedFontsFooter {
  align-items: center;
  background-color: var(--off-white);
  display: flex;
  justify-content: flex-end;
  height: 3.5rem;
}

.SelectedFontsFooter .btn {
  margin: 0 var(--space-lg) 0 0;
}
