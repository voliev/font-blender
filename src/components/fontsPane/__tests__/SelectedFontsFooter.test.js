import React from "react";
import { mount, shallow } from "enzyme";
import SelectedFontsFooter from "../SelectedFontsFooter";

const setup = (renderFn = shallow) => {
  const clearSelectedFontsButtonClickHandler = jest.fn();
  const wrapper = renderFn(
    <SelectedFontsFooter
      clearSelectedFontsButtonClickHandler={
        clearSelectedFontsButtonClickHandler
      }
    />
  );

  return { wrapper, clearSelectedFontsButtonClickHandler };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { wrapper, clearSelectedFontsButtonClickHandler } = setup(mount);

  expect(typeof wrapper.prop("clearSelectedFontsButtonClickHandler")).toBe(
    "function"
  );
  expect(wrapper.prop("clearSelectedFontsButtonClickHandler")).toBe(
    clearSelectedFontsButtonClickHandler
  );
});

it("should call handler on button click", () => {
  const { wrapper, clearSelectedFontsButtonClickHandler } = setup(mount);

  wrapper.find("button").simulate("click");

  expect(clearSelectedFontsButtonClickHandler).toHaveBeenCalledTimes(1);
});
