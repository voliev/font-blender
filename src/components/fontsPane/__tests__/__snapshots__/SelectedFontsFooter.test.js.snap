// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="SelectedFontsFooter"
>
  <button
    className="btn btn-no-background"
    onClick={[MockFunction]}
    title="Clear selected fonts"
    value="clear"
  >
    Clear
  </button>
</div>
`;
